tests/test_middleware.py

```diff
@@ -297,3 +297,27 @@ async def handler(request):
 
     _, response = app.test_client.get("/")
     assert response.json["foo"] == "bar"
+
+
+def test_middleware_return_response(app):
```

Comment:
We need to add some more tests here to cover some of the new edge cases we are solving with this PR.

Reply:
Test cases added, considered as resolved. Some new test cases failed, will try to fix.